modules/security/src/main/java/org/eclipse/xpanse/modules/security/zitadel/ZitadelIdentityProviderService.java

```diff
@@ -172,6 +176,11 @@ public CurrentUserInfo getCurrentUserInfo() {
                     }
                 }
             }
+            OAuth2Token token =
+                    ((AbstractOAuth2TokenAuthenticationToken<?>) authentication).getToken();
+            currentUserInfo.setToken(
+                    ZitadelOauth2Constant.USER_TOKEN_PREFIX + token.getTokenValue());
+            CurrentUserInfoCacheHolder.getSingleton().setCurrentUserInfo(currentUserInfo);
```

Comment:
If current solution using `ConcurrentHashMap` works well,  please move the set action to `IdentityProviderManager`. If change to use the new solution `ConcurrentHashMap`, delete the set acition.